### ISSUE TITLE
Doesn't work for dependencies of dependencies

### ISSUE THREAD

**Issue by donaldpipowitch** [bholloway/omit-tilde-webpack-plugin#5]
``` json
{
  "name": "module-a"
}
```

``` less
// module-a
@coolColor: #f938ab;

p {
  color: @coolColor;
}
```

---

``` json
{
  "name": "module-b",
  "dependencies": {
    "module-a": "v0.1.0"
  }
}
```

``` less
// module-b
@import "module-a/style";
```

---

``` json
{
  "name": "module-c",
  "dependencies": {
    "module-b": "v0.1.0"
  }
}
```

``` less
// module-c
@import "module-b/style";
```

**Error**

```
Module build failed: Cannot resolve 'file' or 'directory' ./module-a/style.less in /Users/foo/Workspace/module-c/node_modules/module-b/src
```

**Only solution so far**

``` json
{
  "name": "module-c",
  "dependencies": {
    "module-a": "v0.1.0", // THIS SHOULD NOT BE NEEDED
    "module-b": "v0.1.0"
  }
}
```

**Comment by bholloway** [comment]
@donaldpipowitch I see your problem.

Have you tried [explicitly specifying](https://github.com/bholloway/omit-tilde-webpack-plugin#peer-projects-and-other-dependencies) those package names. Nasty but does it work?

The problem is that the plugin has to make a decision as to whether the import is intended to be a package or a relative file. It is fair to scan the root `package.json` since (per `resolve.root`) it is usually the primary source of packages.

However it is cool to have packages within packages (per `resolve.modulesDirectories`). However the problem becomes how to find those packages. Would the plugin detect `resolve.modulesDirectories` and enable a hierarchical search? This would be pretty costly.

Another option may be to simply prepend a tilde and resolve on-risk. If that passes then go with it. If it fails then don't use it. Since it uses webpack's own resolve it would be more correct. However it might need more thought.

**Comment by donaldpipowitch** [comment]
> Have you tried explicitly specifying those package names. Nasty but does it work?

Not yet. Maybe I can do that. I probably would need to find a pattern or maybe a flag in the `package.json` of my modules to automatically find modules without `~`.

> Another option may be to simply prepend a tilde and resolve on-risk. If that passes then go with it. If it fails then don't use it.

Or the other way around...? Try to add a tilde, if the file can't be resolved relatively? Because that would be the default behaviour for Less and CSS? (That would probably be like https://github.com/webpack/less-loader/issues/74#issuecomment-209283243.)

**Comment by bholloway** [comment]
I have just published `1.0.0` with the mature code so that #6 can potentially become `2.0.0`.

Implemented as per your suggestion. Performance seems to be a little better than before.

I have tested in an non-complex project so I will need you to check your more complex use-case.

**Comment by donaldpipowitch** [comment]
Thanks, I'll try it.

**Comment by donaldpipowitch** [comment]
Thanks, but sadly it doesn't seem to work :( I'll try to find out why...Thanks, but sadly it doesn't seem to work :( I'll try to find out why...

**Comment by donaldpipowitch** [comment]
It seems to work with `$ npm i --save bholloway/omit-tilde-webpack-plugin#feature/major-rewrite` 😍

**Comment by bholloway** [comment]
Cool thanks.

Sorry I just realised I'm failing another use case so may need to tweak. I'll let you know.

Once I do that and rewrite `readme.md` I can make a `2.0.0` release.

Do you expect that will close out your issue on webpack/less-loader#74 or is there more going on there?

**Comment by donaldpipowitch** [comment]
> Do you expect that will close out your issue on webpack/less-loader#74 or is there more going on there?

Currently I can't think of any more problems I'd encounter. Thank you very much for your help.

**Comment by bholloway** [comment]
Hey @donaldpipowitch I have a final version I am happy with.

Can you please retest and I will publish.

Specifically:
- delete `node_modules/omit-tilde-webpack-plugin`
- `$ npm i --save bholloway/omit-tilde-webpack-plugin#feature/major-rewrite` as before
- retest build

Thanks for your help.

**Comment by donaldpipowitch** [comment]
I'm now on `git://github.com/bholloway/omit-tilde-webpack-plugin.git#26732334b89ba63876c5f8b4e1c58414b2531f72`. Seems to work like it should 👍

**Comment by bholloway** [comment]
excellent.

**Comment by bholloway** [comment]
closed by #6

**Comment by bholloway** [comment]
I've published 2.0.0.

@donaldpipowitch I will leave it to you to comment on webpack/less-loader#74 once you are confident that you have a full solution.

Good working with you.
